lib/models/room.js

```diff
@@ -1120,70 +1120,81 @@ function calculateRoomName(room, userId, ignoreRoomNameEvent) {
     }
 
     // get members that are NOT ourselves and are actually in the room.
-    var members = utils.filter(room.currentState.getMembers(), function(m) {
+    var otherMembers = utils.filter(room.currentState.getMembers(), function(m) {
         return (m.userId !== userId && m.membership !== "leave");
     });
+    var allMembers = utils.filter(room.currentState.getMembers(), function(m) {
```

Comment:
use this to derive `otherMembers` and `myMemberEventArray` (since there might be a large number of `leave` members?)